packages/dds/tree/src/test/tablePerformanceTestUtilities.ts

```diff
@@ -0,0 +1,190 @@
+ * @param cellValue - The initial value to set in each cell of the table.
+ * @returns A fully initialized table tree definition, including table instance, undo/redo stacks, and a cleanup function.
+ */
+export function createTableTree(tableSize: number, cellValue: string): TableTreeDefinition {
+	const sharedTreeFactory = DefaultTestSharedTreeKind.getFactory();
+	const runtime = new MockFluidDataStoreRuntime({
+		idCompressor: createIdCompressor(),
+		attachState: AttachState.Detached,
+	});
+	const tree = sharedTreeFactory.create(runtime, "tree");
+	const treeView = tree.viewWith(
+		new TreeViewConfiguration({
+			schema: Table,
+			enableSchemaValidation: true,
+		}),
+	);
+
+	treeView.initialize(Table.empty());
+	const undoStack: Revertible[] = [];
+	const redoStack: Revertible[] = [];
+
+	function onDispose(disposed: Revertible): void {
+		const redoIndex = redoStack.indexOf(disposed);
+		if (redoIndex !== -1) {
+			redoStack.splice(redoIndex, 1);
+		} else {
+			const undoIndex = undoStack.indexOf(disposed);
+			if (undoIndex !== -1) {
+				undoStack.splice(undoIndex, 1);
+			}
+		}
+	}
+
+	const unsubscribeFromCommitAppliedEvent = treeView.events.on(
+		"commitApplied",
+		(commit, getRevertible) => {
+			if (getRevertible !== undefined) {
+				const revertible = getRevertible(onDispose);
+				if (commit.kind === CommitKind.Undo) {
+					redoStack.push(revertible);
+				} else {
+					undoStack.push(revertible);
+				}
+			}
+		},
+	);
+	const unsubscribe = (): void => {
+		unsubscribeFromCommitAppliedEvent();
+		for (const revertible of undoStack) {
+			revertible.dispose();
+		}
+		for (const revertible of redoStack) {
+			revertible.dispose();
+		}
+	};
+
+	const table = treeView.root;
+	for (let i = 0; i < tableSize; i++) {
+		const column = new Column({ id: `column-${i}` });
+		table.insertColumn({ index: i, column });
+	}
+	for (let i = 0; i < tableSize; i++) {
+		const row = new Row({ id: `row-${i}`, cells: {} });
+		table.insertRow({ index: i, row });
+	}
+	for (let i = 0; i < tableSize; i++) {
+		for (let j = 0; j < tableSize; j++) {
+			table.setCell({
+				key: {
+					column: `column-${i}`,
+					row: `row-${j}`,
+				},
+				cell: { cellValue },
+			});
+		}
+	}
+
+	return {
+		table,
+		undoStack,
+		redoStack,
+		unsubscribe,
+	};
+}
+
+/**
+ * Currently table schema does not support removing cells when a column is removed.
+ * This function provides a way to remove a column and its associated cells from the table. Might remove in the future
+ * if the table schema is updated to handle this automatically.
+ */
+export function removeColumnAndCells(
+	table: InstanceType<typeof Table>,
+	columnId: string,
+): void {
+	Tree.runTransaction(table, () => {
+		const column = table.getColumn(columnId);
+		assert(column !== undefined, `Column with ID "${columnId}" does not exist.`);
+		table.removeColumn(column);
+		for (const row of table.rows) {
+			table.removeCell({ column, row });
+		}
```

Comment:
Just because we want to be perf-conscious in these tests, we might be able to get away with:
```suggestion
		table.removeColumn(columnId);
		for (const row of table.rows) {
			table.removeCell({ column: columnId, row });
		}
```

I believe the above should work - `removeCell` supports specifying either the node or ID for column and row.